sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/models/AttestationOptions.java

```diff
@@ -0,0 +1,157 @@
+     * @return the AttestationOptions object.
+     */
+    AttestationOptions interpretRunTimeDataAsBinary();
+
+    /**
+     * When the attestation service receives the RunTime Data, it should interpret the data as
+     * binary, and return the data in the {@link AttestationResult#getRuntimeClaims()} property
+     * @return the AttestationOptions object.
+     */
+    AttestationOptions interpretRunTimeDataAsJson();
+
+    /**
+     * Set the initTimeData property: Base64Url encoded "InitTime data". The MAA will verify that the init data was
+     * known to the enclave. Note that InitTimeData is invalid for CoffeeLake processors.
+     *
+     * @param initTimeData the initTimeData value to set.
+     * @return the AttestOpenEnclaveRequest object itself.
+     */
+    AttestationOptions setInitTimeData(byte[] initTimeData);
+
+    /**
+     * Retrieves the InitTimeData property to be sent to the service.
+     * @return The InitTimeData value set by {@link AttestationOptions#setInitTimeData}
+     */
+    byte[] getInitTimeData();
+
+    /**
+     * When the attestation service receives the InitTime Data, it should interpret the data as
+     * binary, and return the data in the {@link AttestationResult#getInittimeClaims()} property
+     * @return the AttestationOptions object.
+     */
+    AttestationOptions interpretInitTimeDataAsBinary();
+
+    /**
+     * When the attestation service receives the InitTime Data, it should interpret the data as
+     * binary, and return the data in the {@link AttestationResult#getInittimeClaims()} property
+     * @return the AttestationOptions object.
+     */
+    AttestationOptions interpretInitTimeDataAsJson();
+
+    /**
+     * Set the draftPolicyForAttestation property: Attest against the provided draft policy. Note that the resulting
+     * token cannot be validated.
+
+     * Note that the resulting token cannot be validated.
+     *
+     * @param draftPolicyForAttestation the draftPolicyForAttestation value to set.
+     * @return the AttestOpenEnclaveRequest object itself.
+     */
+    AttestationOptions setDraftPolicyForAttestation(String draftPolicyForAttestation);
+
+    /**
+     * Gets the draftPolicyForAttestation property which is used to attest against the draft policy.
+     *
+     * Note that the resulting token cannot be validated.
+     *
+     * @return The draft policy if set.
+     */
+    String getDraftPolicyForAttestation();
+
+    /**
+     * Validates the instance.
+     *
+     * If the caller has called setRunTimeData without calling interpretRunTimeDataAsBinary or
+     * interpretRunTimeDataAsJson, this will throw an exception, similarly if you call setInitTimeData
+     * without also calling either interpretInitTimeDataAsJson or interpretInitTimeDataAsBinary.
+     *
+     * @throws IllegalArgumentException thrown if the instance is not valid.
+     */
+    void validate();
```

Comment:
You are likely going  to be asked in the arch board review about this. We prefer not to validate inputs.

Reply:
???? That is weird, since this is EXACTLY what the generated code does. I'm literally copying the pattern which is used by autorest.

And this validation ONLY checks for states which cannot be expressed at runtime - these checks turn null pointer exceptions into human readable exceptions.

IMHO, a human readable explanatory exception is always better than a null pointer exception where the customer needs to reverse engineer the cause of the exception.

Reply:
I didn't mean this is not correct for you to use. I just meant you will be asked what kind of validations are being done here.